test/main.cpp

```diff
@@ -0,0 +1,4 @@
+#define CATCH_CONFIG_MAIN
+#include "catch2/catch.hpp"
+
+/// DO NOT MODIFY
```

Comment:
seems like an autogenerated file, is it something we need in the repo?

Reply:
This is the main test runner - I've used this since it increases compile times by a decent margin